### PR TITLE
remove unnecessary header fields

### PR DESCRIPTION
**Pull request by tudor-malene**
### Why this change is needed

we currently have unnecessary fields in the rollup and batch header.

### What changes were made as part of this PR

- removes the unnecessary fields
- this included the logic around the bloom filter
- dedups the logic around the rollup/batch signature check

### PR checks pre-merging

Please indicate below by ticking the checkbox that you have read and performed the required
[PR checks](https://github.com/obscuronet/obscuro-internal/blob/main/dev-ops-docs/dev-pr-checks.md)

- [x] PR checks reviewed and performed

### CODE REVIEW

File: go/enclave/components/batch_producer.go [hunk 1]
```diff
@@ -62,7 +62,7 @@ func (bp *batchProducerImpl) ComputeBatch(context *BatchExecutionContext) (*Comp
 	}
 
 	// Create a new batch based on the fromBlock of inclusion of the previous, including all new transactions
-	batch := core.DeterministicEmptyBatch(context.Creator, parent, block, context.Randomness, context.AtTime)
+	batch := core.DeterministicEmptyBatch(parent, block, context.Randomness, context.AtTime)
```
Comment (StefanIliev545) [inline go/enclave/components/batch_producer.go:65]: Why are we removing the sequencer address per batch?

**Review by StefanIliev545 — APPROVED** [review]
Looks ok, but removing the sequencer address from the batches and making it implicit is a bit confusing.

File: go/enclave/components/batch_producer.go (hunk 1 above)
Comment (tudor-malene) [reply go/enclave/components/batch_producer.go:65]: the signature already contains that information

Comment (tudor-malene) [reply go/enclave/components/batch_producer.go:65]: we're not even using that.
We have the approved sequencer stored in the DB and check the signature against that key.

**Pull request merged**